File: WorkIt/app/src/main/java/prot3ct/workit/views/edit_task/base/EditTaskContract.java
package prot3ct.workit.views.edit_task.base;

import prot3ct.workit.base.BaseView;
import prot3ct.workit.view_models.TaskDetailViewModel;

public interface EditTaskContract {
    interface View extends BaseView<EditTaskContract.Presenter> {
        void showListJobsActivity();

        void updateTask(TaskDetailViewModel task);

        void notifySuccessful();

        void showDialogforLoading();

        void dismissDialog();

        void notifyError(String errorMessage);
    }

    interface Presenter {
        void updateTask(int taskId, String title, String startDate, String length,
                        String description, String city, String address, String reward);

        void getTaskDetails(int taskId);
    }
}

File: WorkIt/app/src/main/java/prot3ct/workit/views/edit_task/base/EditTaskContract.kt
package prot3ct.workit.views.edit_task.base

import prot3ct.workit.base.BaseView
import prot3ct.workit.view_models.TaskDetailViewModel

interface EditTaskContract {
    interface View : BaseView<Presenter> {
        fun showListJobsActivity()
        fun updateTask(task: TaskDetailViewModel)
        fun notifySuccessful()
        fun showDialogforLoading()
        fun dismissDialog()
        fun notifyError(errorMessage: String)
    }

    interface Presenter {
        fun updateTask(
            taskId: Int, title: String, startDate: String, length: String,
            description: String, city: String, address: String, reward: String
        )

        fun getTaskDetails(taskId: Int)
    }
}